extension/src/background.ts

```diff
@@ -67,27 +73,57 @@ chrome.alarms.onAlarm.addListener(async (alarm: chrome.alarms.Alarm) => {
 chrome.runtime.onMessage.addListener(async (message: Message, sender, sendResponse) => {
     // console.debug('MESSAGE RECEIVED!', message);
 
-    if (message.type === 'keep-alive') {
-        // console.debug('Received keep-alive message.');
-    } else if (message.type === 'index') {
-        await executeIndexer();
-    } else if (message.type === 'watch') {
-        await runWatcher();
-    } else if (message.type === 'network') {
-        await networkStatusWatcher();
-    } else if (message.type === 'activated') {
-        // console.log('THE UI WAS ACTIVATED!!');
-        // When UI is triggered, we'll also trigger network watcher.
-        await networkStatusWatcher();
+  if (message.type === 'keep-alive') {
+    // console.debug('Received keep-alive message.');
+  } else if (message.type === 'index') {
+    await executeIndexer();
+  } else if (message.type === 'watch') {
+    await runWatcher();
+  } else if (message.type === 'network') {
+    await networkStatusWatcher();
+  } else if (message.type === 'activated') {
+    // console.log('THE UI WAS ACTIVATED!!');
+    // When UI is triggered, we'll also trigger network watcher.
+    await networkStatusWatcher();
+  } else if (message.type === 'getpublickey') {
+
+    if (walletStore == null) {
+      walletStore = new WalletStore();
+      await walletStore.load();
+    }
+
+    var wallets = walletStore.getWallets();
+
+    var res: any;
+    for (let i = 0; i < wallets.length; i++) {
+      const wallet = wallets[i];
+      for (let j = 0; j < wallet.accounts.length; j++) {
+        const account = wallet.accounts[j];
+        res = account.xpub
```

Comment:
Never share xpub outside of the wallet.

Reply:
yeah sure, I didn't have access to the hd class so this is just an example